include/oneapi/dpl/experimental/kt/single_pass_scan.h

```diff
@@ -0,0 +1,435 @@
+    static_assert(oneapi::dpl::unseq_backend::__has_known_identity<_BinaryOp, _Type>::value,
+                  "Only binary operators with known identity values are supported");
+
+    assert("This device does not support 64-bit atomics" &&
+           (sizeof(_Type) < 64 || __queue.get_device().has(sycl::aspect::atomic64)));
+
+    // Next power of 2 greater than or equal to __n
+    auto __n_uniform = ::oneapi::dpl::__internal::__dpl_bit_ceil(__n);
+
+    // Perform a single-work group scan if the input is small
+    if (oneapi::dpl::__par_backend_hetero::__group_scan_fits_in_slm<_Type>(__queue, __n, __n_uniform))
+    {
+        return oneapi::dpl::__par_backend_hetero::__parallel_transform_scan_single_group(
+            oneapi::dpl::__internal::__device_backend_tag{},
+            oneapi::dpl::execution::__dpl::make_device_policy<typename _KernelParam::kernel_name>(__queue),
+            std::forward<_InRange>(__in_rng), std::forward<_OutRange>(__out_rng), __n,
+            oneapi::dpl::__internal::__no_op{}, unseq_backend::__no_init_value<_Type>{}, __binary_op,
+            std::true_type{});
+    }
+
+    constexpr std::size_t __workgroup_size = _KernelParam::workgroup_size;
+    constexpr std::size_t __data_per_workitem = _KernelParam::data_per_workitem;
+
+    // Avoid non_uniform n by padding up to a multiple of workgroup_size
+    std::size_t __elems_in_tile = __workgroup_size * __data_per_workitem;
+    std::size_t __num_wgs = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __elems_in_tile);
+
+    constexpr int __status_flag_padding = SUBGROUP_SIZE;
+    std::size_t __status_flags_size = __num_wgs + 1 + __status_flag_padding;
+
+    std::size_t __mem_align_pad = sizeof(_Type);
+    std::size_t __status_flags_bytes = __status_flags_size * sizeof(_FlagStorageType);
+    std::size_t __status_vals_full_offset_bytes = __status_flags_size * sizeof(_Type);
+    std::size_t __status_vals_partial_offset_bytes = __status_flags_size * sizeof(_Type);
+    std::size_t __mem_bytes = __status_flags_bytes + __status_vals_full_offset_bytes +
+                                __status_vals_partial_offset_bytes + __mem_align_pad;
+
+    std::byte* __device_mem = reinterpret_cast<std::byte*>(sycl::malloc_device(__mem_bytes, __queue));
+    if (!__device_mem)
+        throw std::bad_alloc();
+
+    _FlagStorageType* __status_flags = reinterpret_cast<_FlagStorageType*>(__device_mem);
+    std::size_t __remainder = __mem_bytes - __status_flags_bytes;
+    void* __vals_base_ptr = reinterpret_cast<void*>(__device_mem + __status_flags_bytes);
+    void* __vals_aligned_ptr = std::align(std::alignment_of_v<_Type>, __status_vals_full_offset_bytes,
+                                            __vals_base_ptr, __remainder);
+    _Type* __status_vals_full = reinterpret_cast<_Type*>(__vals_aligned_ptr);
+    _Type* __status_vals_partial = reinterpret_cast<_Type*>(__status_vals_full + __status_vals_full_offset_bytes);
```

Comment:
Address arithmetic fix:

```suggestion
    _Type* __status_vals_partial = reinterpret_cast<_Type*>(__status_vals_full +
                                                            __status_vals_full_offset_bytes / sizeof(_Type));
```
There should've been a number of elements rather than a number of bytes.